lib/player.js

```diff
@@ -990,6 +1072,42 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     return this.wrapWalkerListenersWithPromise_(events);
   }
 
+
+  /**
+   * Tell the player to use <code>canvas</code> for all <code>load</code>
+   * requests until <code>detach</code> or <code>destroy</code> are called.
+   *
+   * <p>
+   * Calling <code>attachCanvas</code> will tell the player to set canvas for
+   * lcevc decoding
+   *
+   * @param {!HTMLCanvasElement} canvas
+   * @return {!Promise}
+   * @export
+   */
+  attachCanvas(canvas) {
+    // Do not allow the player to be used after |destroy| is called.
+    if (this.loadMode_ == shaka.Player.LoadMode.DESTROYED) {
+      return Promise.reject(this.createAbortLoadError_());
+    }
+
+    this.lcevcCanvas_ = canvas;
+
+    // Do not allow this route to be interrupted because calls after this attach
+    // call will depend on the media element being attached.
+    const events = this.walker_.startNewRoute((currentPayload) => {
+      return {
+        node: this.attachNode_,
+        payload: shaka.Player.createEmptyPayload_(),
+        interruptible: false,
+      };
+    });
+
+    // List to the events that can occur with our request.
+    events.onStart = () => shaka.log.info('Starting Canvas attach...');
+    return this.wrapWalkerListenersWithPromise_(events);
```

Comment:
I don't think any of this belongs.  Just set the canvas element.  This shouldn't affect the load graph.

Reply:
Removed this and only canvas setting is done in this function.